### PR TITLE
Update the ingress's hostname if globalDNS FQDN is changed

### PR DESCRIPTION
**Pull request by prachidamle**
This is the fix to update the ingress's hostname and update endpoints if the globalDNS that created the ingress is updated to use a different FQDN. 

https://github.com/rancher/rancher/issues/17832

### CODE REVIEW

**Comment by prachidamle** [comment]
@alena1108 @kinarashah can you review pls.

File: pkg/controllers/management/globaldns/globaldns_handler.go [hunk 1]
```diff
@@ -196,20 +196,28 @@ func (n *GDController) generateNewIngressSpec(globaldns *v3.GlobalDNS) *v1beta1.
...
-	ingress.Status.LoadBalancer.Ingress = n.sliceToStatus(endpoints)
-	updatedObj, err := n.ingresses.UpdateStatus(ingress)
+		if err != nil {
+			return fmt.Errorf("GlobalDNSController: Error updating Ingress %v", err)
+		}
+	}
 
-	if err != nil {
-		return fmt.Errorf("GlobalDNSController: Error updating Ingress %v", err)
+	if len(ingress.Spec.Rules) > 0 {
```
Comment (alena1108) [inline pkg/controllers/management/globaldns/globaldns_handler.go]: when would len of ingress.spec.Rules be 0? and do we have to handle it?

Comment (prachidamle) [reply pkg/controllers/management/globaldns/globaldns_handler.go]: We create the ingress here https://github.com/rancher/rancher/blob/b78355467e57d8103a20a25bbe455dc60ef4b6b4/pkg/controllers/management/globaldns/globaldns_handler.go#L90 and set the ingress.spec.Rules at that time and the hostname as well. 

This is method where we make the check is called when the ingress is already existing, only for updates. So  ingress.Spec.Rules won't be empty really, but adding the check for validation and index checks.

**Comment by alena1108** [comment]
LGTM

**Pull request merged**